WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorViewModel.swift

```diff
@@ -301,6 +376,68 @@ private extension ProductSelectorViewModel {
     }
 }
 
+// MARK: - Multiple selection support
+private extension ProductSelectorViewModel {
+    /// Toggles the selection of the specified product.
+    ///
+    func toggleSelection(productID: Int64) {
+        if selectedProductIDs.contains(productID) {
+            selectedProductIDs.removeAll(where: { $0 == productID })
+        } else {
+            selectedProductIDs.append(productID)
+        }
+    }
+
+    /// Update selected product and variation IDs from initial selected items
+    ///
+    func updateSelectionsFromInitialSelectedItems() {
+        guard initialSelectedItems.isNotEmpty else {
+            return
+        }
+        for id in initialSelectedItems {
+            if products.contains(where: { $0.productID == id }) {
+                selectedProductIDs.append(id)
+            } else {
+                selectedProductVariationIDs.append(id)
+            }
+        }
+    }
```

Comment:
Right now, because `updateSelectionsFromInitialSelectedItems()` is called in `updateProductsResultsController()`, which is called any time a sync or search action is performed, the initial selected item IDs are being added repeatedly to `selectedProductIDs` or `selectedProductVariationIDs`. I think it makes the most sense to only do this update once, when the view model is first initialized, rather than within `updateProductsResultsController()`. Alternately, we should make sure the `id` isn't already contained in the array before appending it.

You can see the effect as the count goes up in the "Select # Products" button, with only 1 product selected:

https://user-images.githubusercontent.com/8658164/164703111-287eba68-0e68-4ebe-8d01-6868da392cb9.mp4

Reply:
Good point! I think it's still necessary that the selections are updated when products change - this is important when the selected item is not on the first result page. I added checks to make sure the IDs have not been added to the selected lists though - this should fix the incorrect count of items.